Tools/AP_Periph/can.cpp

```diff
@@ -95,11 +96,17 @@ static uint8_t transfer_id;
 #define CAN_PROBE_CONTINUOUS 0
 #endif
 
+
+AP_CANManager::Driver_Type AP_Periph_FW::can_protocol[HAL_NUM_CAN_IFACES];
 #if CONFIG_HAL_BOARD == HAL_BOARD_CHIBIOS
-static ChibiOS::CANIface can_iface[HAL_NUM_CAN_IFACES];
+ChibiOS::CANIface* AP_Periph_FW::can_iface_periph[HAL_NUM_CAN_IFACES];
 #elif CONFIG_HAL_BOARD == HAL_BOARD_SITL
-static HALSITL::CANIface can_iface[HAL_NUM_CAN_IFACES];
+HALSITL::CANIface* AP_Periph_FW::can_iface_periph[HAL_NUM_CAN_IFACES];
 #endif
+
+AP_HAL::CANIface *const (*const CANSensor::can_iface_periph)[HAL_NUM_CAN_IFACES] = (AP_HAL::CANIface* (*)[HAL_NUM_CAN_IFACES])&AP_Periph_FW::can_iface_periph;
```

Comment:
this is awful - I was expecting the init function to do something like set_can_iface() on CANSensor as a static function